src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/GetWorkerSteps.java

```diff
@@ -73,34 +111,154 @@ public void tearDownGatewayEventMonitor() throws IOException {
 
         List<AdeccoResponse> adeccoResponseList = new ArrayList<>();
         adeccoResponseList.add(adeccoResponse);
         adeccoMockUtils.addUsersAdecco(adeccoResponseList);
+
+
+        // Write code here that turns the phrase above into concrete actions
+        //throw new cucumber.api.PendingException();
     }
 
-    @Then("as FSDR system I can pull off Adecco")
-    public void asFSDRSystemICanPullOffAdecco() throws IOException {
+
+    @When("FSDR pulls data from Adecco")
+    public void fsdr_pulls_data_from_Adecco() throws IOException {
         fsdrUtils.ingestAdecco();
+        fsdrUtils.ingestRunFSDRProcess();
         boolean hasBeenTriggered = gatewayEventMonitor.hasEventTriggered("<N/A>", INGEST_FROM_ADECCO, 10000L);
         assertTrue(hasBeenTriggered);
+
+        // Write code here that turns the phrase above into concrete actions
+
+    }
+
+    @When("employee assigned device temporary fix do not use this code")
+    public void fsdrPullsDevicesFromXMA() {
+        //Todo this is a temporary fix so that an employee will have a device for the LWS extract,
+        // this needs to be replaced with an xma get devices mock
+        fsdrUtils.postDeviceToFsdr();
     }
 
-    @And("search database for {string} employee with ID {string}")
-    public void searchDatabaseForEmployeeWithID(String source, String employeeId) {
+    @Then("Check the Employee created in FSDR database with ID {string}")
+    public void check_the_Employee_created_in_FSDR_database_with_ID(String  employeeId) {
+
         String responseEmployeeId;
-        ResponseEntity<AdeccoResponseList> results = adeccoMockUtils.getEmployeeById(employeeId);
+        ResponseEntity<Employee> employeeResponseEntity = fsdrUtils.retrieveEmployee(employeeId);
+        assertEquals(HttpStatus.OK, employeeResponseEntity.getStatusCode());
+        fsdrEmployee = employeeResponseEntity.getBody();
+
+        responseEmployeeId = fsdrEmployee.getUniqueEmployeeId();
+
+        assertEquals(employeeId, responseEmployeeId);
+        // Write code here that turns the phrase above into concrete actions
+
+    }
+
+    @Then("FSDR update the external systems")
+    public void fdr_update_the_external_system() throws IOException {
+
+        fsdrUtils.ingestGsuit();
+        boolean hasBeenTriggered = gatewayEventMonitor.hasEventTriggered(fsdrEmployee.getUniqueEmployeeId(), GSUITE_COMPLETE, 10000L);
+        assertTrue(hasBeenTriggered);
+
+        fsdrUtils.ingestXma();
+       boolean hasBeenTriggeredxma = gatewayEventMonitor.hasEventTriggered(fsdrEmployee.getUniqueEmployeeId(), XMA_EMPLOYEE_SENT, 10000L);
+        assertTrue(hasBeenTriggeredxma);
+
+        fsdrUtils.ingestSnow();
+        boolean hasBeenTriggeredsnow = gatewayEventMonitor.hasEventTriggered(fsdrEmployee.getUniqueEmployeeId(), SERVICENOW_CREATED, 10000L);
+        assertTrue(hasBeenTriggeredsnow);
+
+        fsdrUtils.ingestGranby();
```

Comment:
this and the following calls should wait for the appropriagte event. Otherwise we will get a race condition, especially when we go to event driven

Edit: Ive just looked further down, you check for the event down there is that correct? if so shouldwe do it for the calls above?